diff --git a/node_modules/better-auth/dist/api/index.mjs b/node_modules/better-auth/dist/api/index.mjs
index 112cdb4..8808f00 100644
--- a/node_modules/better-auth/dist/api/index.mjs
+++ b/node_modules/better-auth/dist/api/index.mjs
@@ -751,7 +751,8 @@ const router = (ctx, options) => {
     ],
     async onRequest(req) {
       const disabledPaths = ctx.options.disabledPaths || [];
-      const path = new URL(req.url).pathname.replace(basePath, "");
+      const path = new URL(req.url, process.env.BASE_URL || "http://localhost:3000")
+        .pathname.replace(basePath, "");
       if (disabledPaths.includes(path)) {
         return new Response("Not Found", { status: 404 });
       }
